### PR TITLE
Debugging choose_impl parallel issue

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/kernel_selector/core/kernel_base.cpp [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 namespace kernel_selector {
 const primitive_db KernelBase::db;
-thread_local size_t KernelBase::counter = 0;
+// thread_local size_t KernelBase::counter = 0;
+std::atomic<size_t> KernelBase::counter(0);
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/kernel_base.cpp:12]: now that we are using processing order as unique id, we no longer need this counter isnt it?

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp [hunk 2]
```diff
@@ -11,17 +11,95 @@
...
+#else
+    // std::cout << "CLDNN_THREADING_SEQ" << std::endl;
     for (auto& node : p.get_processing_order()) {
+        count++;
+        auto idx = p.get_processing_order().get_processing_number(node) - 1;
+        node->set_unique_id(idx);
         if (!node->is_type<internal_primitive>() && !node->is_type<data>()) {
             node->get_output_layout();
             if (!node->is_type<data>() && !(node->is_type<mutable_data>() && node->get_dependencies().empty())) {
+                real_count++;
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp:96]: this is to be removed right?

**Comment by yeonbok** [comment]
I think there will be one more problem for unique id for building body network of loop operation, 
because the processing order will start from 0 again for body network.. 
i.e., if same kernel is being used from outer network and body network, the same id might be used again. 
(very low probability to have same processing order but it might be !) 

I think we need to consider about this..hmm

**Comment by yeonbok** [comment]
How about following ?
1) Modify the type of "id" to be string instead of size_t 
2) append id as "[network_id]_[processing_order]"

**Comment by ahnyoung-paul** [comment]
> seems that this part can be shared b/w parallel and sequential codes

Sorry I check this review at now. I will update it at the official PR. thanks.

**Comment by ahnyoung-paul** [comment]
@yeonbok's last comments are applied to the merged PR. thanks.

**Pull request closed without merging**